core/server/utils/image-size-from-url.js

```diff
@@ -51,71 +48,56 @@ module.exports.getImageSizeFromUrl = function getImageSizeFromUrl(imagePath) {
             }
         }
 
-        options = url.parse(imagePath);
+        imagePath = url.parse(imagePath);
+        requestOptions = {
+            headers: {
+                'User-Agent': 'Mozilla/5.0'
+            },
+            timeout: timeout,
+            encoding: null
+        };
 
-        requestHandler = imagePath.indexOf('https') === 0 ? https : http;
-        options.headers = {'User-Agent': 'Mozilla/5.0'};
+        got(
+            imagePath,
+            requestOptions
+        ).then(function (response) {
+            try {
+                dimensions = sizeOf(response.body);
```

Comment:
I tried refactoring as you recommended, but I couldn't make it work... Guess I need some more time or assistance with this 😔